compiler/cpp/src/thrift/generate/t_generator.h

```diff
@@ -96,7 +97,47 @@ class t_generator {
     return escape_string(constval->get_string());
   }
 
+  /**
+   * Check if all identifiers are valid for the target language
+   */
+  virtual void validate_input() const;
+
 protected:
+  virtual std::set<std::string> lang_keywords() {
+    std::string keywords[] = { "BEGIN", "END", "__CLASS__", "__DIR__", "__FILE__", "__FUNCTION__",
+        "__LINE__", "__METHOD__", "__NAMESPACE__", "abstract", "alias", "and", "args", "as",
+        "assert", "begin", "break", "case", "catch", "class", "clone", "continue", "declare",
```

Comment:
Should we include C++11 keywords like auto, constexpr, nothrow?

https://en.cppreference.com/w/cpp/keyword

Reply:
I don't think it's safe to add keywords to this keyword list, because it will add restrictions to all the languages (except for py) and break things. It's possible to add a complete list of cpp keywords to cpp generator.